idx/memory/memory.go

```diff
@@ -980,7 +980,7 @@ func (m *MemoryIdx) Delete(orgId int, pattern string) ([]idx.Archive, error) {
 	}
 
 	for _, f := range found {
-		deleted := m.delete(orgId, f, true)
+		deleted := m.delete(orgId, f, true, true)
```

Comment:
is this the behavior we want? let's say you have data under these 2 paths:
```
foo.bar
foo.bar.baz
```
afaik a render query for `foo.*` would only return the first one, so a delete should probably also only delete the first one.

IOW is there even a use case at all for recursive deletes?

Reply:
if you want a recursive delete then `foo.**` should match them all, so I'd tend to agree that the user should be required to be explicit.

Reply:
> if you want a recursive delete then foo.** should match them all

is there any existing implementation/use of this format? it seems sensible.

Reply:
It's not documented in https://graphite.readthedocs.io/en/latest/render_api.html#target (I'll have to fix that) but is supported by the standard whisper/rrd finder in graphite-web https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/finders/standard.py#L97